numba/core/utils.py

```diff
@@ -532,8 +532,6 @@ def unified_function_type(numba_types, require_precise=True):
             if isinstance(t, types.UndefinedFunctionType):
                 if undefined_function is None:
                     undefined_function = t
-                else:
-                    assert undefined_function == t
```

Comment:
I'm trying to understand this patch. I see that when this condition is reached, this function will return None. Couldn't it return `None` at this point?